src/components/select/CustomSelectDropdown.tsx

```diff
@@ -0,0 +1,196 @@
+import { SearchBox } from "@canonical/react-components";
+import {
+  FC,
+  KeyboardEvent,
+  LiHTMLAttributes,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import classnames from "classnames";
+
+export type CustomSelectOption = LiHTMLAttributes<HTMLLIElement> & {
```

Comment:
aah yes, we should destruct `option` into the `li` elements. Also I removed the `title` field from the `CustomSelectOption` type since it's already part of `LiHTMLAttributes`